src/data/equiv/one_sided.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import data.set.function
+
+/-!
+# Functions with a bundled one-sided inverse
+
+This provides `α ↬ β` (`injection`) and `α ↠ β` (`surjection`), for functions associated with a
+specific left- or right-inverse. For the two-sided inverse, use `α ≃ β` (`equiv`).
+
+For cases when you care only that a function is injective, and do not need a specific inverse, use
+`α ↪ β` (`embedding`).
```

Comment:
Do you want to also mention `monomorphism` and `epimorphisms` here?

Reply:
No because I don't know anything about how to use those. Perhaps @b-mehta can suggest some wording.

Reply:
I don't think it's necessary to add, but this is essentially giving bundled split monomorphisms `↬`, bundled split epimorphisms `↠` and contrasting them with `↪` which correspond to just monomorphisms (all in the category of types of course).

Reply:
Would `split_monomorphism` and `split_epimorphism` be better names than `injection` and `surjection`?

Reply:
I think they're less well-known and so might be harder to search for